### PR TITLE
feat(cb2-4684): create new v2 getTechRecords function

### PR DESCRIPTION
**Pull request by aaron-downing**
## Create v2 getTechRecords endpoint

This PR just covers the basic functionality of the V2 getTechRecords endpoint 

**_T.B.C will be another/few PRs to add extra unit/integration testing as well as the API routing before merging to develop, this was just getting slightly large...._**

- Adds new v2 getTechRecords endpoint which retrieves FLAT vehicle technical records from the `flat-tech-records` DynamoDB and then transforms them into the legacy structure that can be found in the `technical-records` DynamoDB.
- Adds `flat-tech-records` DynamoDB to local Serverless development with seed data.
- Refactors existing codebase to use dependency injection and keep things DRY
- Reviewed existing unit test structure to make it more legible what each unitTest file relates to
- Fixed an existing unit test in TechRecordsService.unitTest.ts which was using cloneDeep incorrectly and practically asserting on `result = result`

[link to ticket number](https://dvsa.atlassian.net/browse/CB2-4684)

## Checklist

- [X] Code has been tested manually
- [X] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by Ian-Hodges — APPROVED** [review]
Code base needs a run through with prettier :-)

**Comment by aaron-downing** [comment]
> Code base needs a run through with prettier :-)



> Code base needs a run through with prettier :-)

**Comment by aaron-downing** [comment]
> Code base needs a run through with prettier :-)

Only tslint is installed on this repo, will create a task to install eslint/prettier

**Review by kpearmanbjss — APPROVED** [review]

File: src/models/TechRecordsDAO.ts [hunk 1]
```diff
@@ -46,66 +97,17 @@ class TechRecordsDAO {
     };
 
     if (isSysNumSearch(searchCriteria)) {
-      // Query for a specific System Number
```
Comment (tom-bjss) [inline src/models/TechRecordsDAO.ts:49]: Really nice refactoring here Aaron 👍

**Pull request merged**